packages/teams-js/src/public/interfaces.ts

```diff
@@ -180,7 +180,34 @@ export enum FileOpenPreference {
  * Represents the structure of the received context message.
  */
 
+export enum ActionObjectType {
+  M365Conontent = 'm365content',
+}
+
+export interface BaseActionObject<T extends ActionObjectType> {
```

Comment:
I think we need the M365Content type to be defined here, and then we should be testing it in the E2E test to make sure the data makes it down to the application from the host

Reply:
isn't the M365 content type above this interface?  Also,  you're exactly right, I plan on adding the e2e tests after this PR is merged!

Reply:
do you mean that we should not have the generic and we should _only_ have the M365Content type?  I wanted to keep these as flexible as possible since we don't know the types of objects we could add in the future

Reply:
Followed up with you offline. E2E tests after this merge sounds perfect, I mis-spoke and meant unit tests (app.spec.ts). You'll also need to add the M365ContentAction interface to this file.

Reply:
ok after speaking I realized I missed some items from the other PR.  I needed to specify thee actual content type.  I added these missing interfaces and a typeguard so that we can identify these from each other in the future